profiles-jar-examples/main-app/src/main/java/com/c12e/cortex/examples/JoinMultiDatasource.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2022 Cognitive Scale, Inc. All Rights Reserved.
+ *
+ *  See LICENSE.txt for details.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.c12e.cortex.examples;
+
+import com.c12e.cortex.phoenix.profiles.spark.FabricSession;
+import com.c12e.cortex.phoenix.profiles.spark.client.LocalSecretClient;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SaveMode;
+import org.apache.spark.sql.SparkSession;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@CommandLine.Command(name = "sql-join-conns", description = "Example Joining Connections", mixinStandardHelpOptions = true)
+public class JoinMultiDatasource extends  BaseCommand implements Runnable {
+    @CommandLine.Option(names = {"-p", "--project"}, description = "Project to use", required = true)
+    String project;
+
+    @CommandLine.Option(names = {"-c", "--config"}, description = "Ingestion config file path", required = true)
+    String configFilePath;
+
+    @CommandLine.Spec
+    private CommandLine.Model.CommandSpec cmdSpec;
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    /**
+     * Code to run for a command
+     */
+    @Override
+    public void run() {
+        //get spark session
+        SparkSession session = getSparkSession(getDefaultProps());
+        session.conf().getAll().toStream().print();
+
+        //create fabric session
+        LocalSecretClient.LocalSecrets localSecrets = new LocalSecretClient.LocalSecrets();
+        FabricSession fabricSession = getFabricSession(session, localSecrets);
+
+        HashMap config;
+        try {
+            JsonNode json = mapper.readTree(Paths.get(configFilePath).toFile());
+            config = mapper.convertValue(json, HashMap.class);
+            ((Map)config.getOrDefault("variables", Collections.emptyMap())).forEach((k, v) -> session.sql(String.format("set %s=%s", k, v)));
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(String.format("Ingestion confile file %s must be valid JSON file", configFilePath), e);
+        } catch (IOException e) {
+            throw new RuntimeException(String.format("Config file path %s not found", configFilePath), e);
+        }
+        String queryTemplate = (String) ((Map)config.get("join")).get("query");
```

Comment:
Think it would be nice example wise if this config loading was wrapped in separate methods, too kinda hide some of the loading and type casting. (Reduce noise from usage of profile library)